### PR TITLE
Testability Statements 5.5.4, 5.5.4.1, 5.6 and 5.7

### PR DESCRIPTION
**Pull request by Therecanbeonlyone1969**
Also made some minor clarifying edits to some requirements.

### CODE REVIEW

**Comment by Therecanbeonlyone1969** [comment]
The next PR will be for 5.5.4.2 which contains a lot of requirements, about 70. However, the requirements are very similar in structure and therefore the testability statements will be very similar.

File: core/baseline-core-v1.0-psd01.md [hunk 1]
```diff
@@ -1692,65 +1692,332 @@ First, the document lists the requirements common to mono-directional BPI servic
...
-The zero-knowledge proof of correctness of a state object utilized in one or more mono-directional and/or bi-/multi-directional BPI Services MUST be succinct.
+The non-interactive zero-knowledge proof of correctness of a BPI state object utilized in one or more mono-directional and/or bi-/multi-directional BPI Services MUST be succinct.
 
 Succinct in the context of zero-knowledge proofs means that a zero-knowledge proof is verifiable by any 3rd party in a time that is sublinear to the size of the prover system that generated the proof.
 
+[[R124]](#r125) Testability:
+
+Preconditions:
+
+* A non-interactive zero-knowledge proof of a state object utilized in one or more mono-directional and/or bi-/multi-directional BPI Service from a prover system that can generate a zero-knowledge proof of correctness for the output of size N. The size N is given by the size of the circuit used to generate the proof in terms of the number of circuit constraint equations.
```
Comment (Ybittan) [inline core/baseline-core-v1.0-psd01.md:1726]:
```suggestion
* A non-interactive zero-knowledge proof of a state object utilized in one or more mono-directional and/or bi-/multi-directional BPI Services from a prover system that can generate a zero-knowledge proof of correctness for the output of size N. The size N is given by the size of the circuit used to generate the proof in terms of the number of circuit constraint equations.
```

File: core/baseline-core-v1.0-psd01.md [hunk 2]
```diff
@@ -1692,65 +1692,332 @@ First, the document lists the requirements common to mono-directional BPI servic
...
+
+Test Steps:
+
+1. Verify the zero-knowledge proof using the verifier system.
+2. Measure the number of computational steps taken by the verifier system to verify the proof.
+3. Verify that the number of computational steps taken to verify the proof is proportional at most to or less than the poly-logarithm of the size of the prover system.
+
+Test Passing Criteria:
+
+* The test will pass if the non-interactive zero-knowledge proof of a state object utilized in one or more mono-directional and/or bi-/multi-directional BPI Service can be verified by the verifier system in a number of computational steps at most proportional to the size of the prover system that generated the proof.
```
Comment (Ybittan) [inline core/baseline-core-v1.0-psd01.md:1737]:
```suggestion
* The test will pass if the non-interactive zero-knowledge proof of a state object utilized in one or more mono-directional and/or bi-/multi-directional BPI Services can be verified by the verifier system in a number of computational steps at most proportional to the size of the prover system that generated the proof.
```

File: core/baseline-core-v1.0-psd01.md [hunk 3]
```diff
@@ -2258,14 +2525,41 @@ The detailed API specification of the Mono-directional and Bi-/Multi-directional
...
 Data in transit between BPIs MUST be encrypted.
 
+[[R209]](#r209) Testability:
+
+Test Preconditions:
+* BPI communication infrastructure is set up and operational.
+* Cryptographic keys for encryption and decryption are available and properly configured.
+
+Test Steps:
+1. The test system initiates a communication between two BPIs (source and target).
```
Comment (Ybittan) [inline core/baseline-core-v1.0-psd01.md:2544]:
```suggestion
1. The test system initiates communication between two BPIs (source and target).
```

**Review by Ybittan — CHANGES_REQUESTED** [review]
Other than a few minor grammatical suggestions, LGTM.

**Review by Kasshern — APPROVED** [review]

**Pull request merged**